DigitalLearningSolutions.Web/Controllers/TrackingSystem/CentreConfiguration/CentreConfigurationController.cs

```diff
@@ -131,5 +137,133 @@ public IActionResult EditCentreWebsiteDetails(EditCentreWebsiteDetailsViewModel
+                ModelState.AddModelError(nameof(EditCentreDetailsViewModel.CentreLogoFile),
+                    "Preview your new centre logo before saving");
+            }
+
+            if (!ModelState.IsValid)
+            {
+                return View(model);
+            }
+
+            var centreId = User.GetCentreId();
+
+            centresDataService.UpdateCentreDetails(
+                centreId,
+                model.NotifyEmail,
+                model.BannerText!,
+                model.CentreSignature,
+                model.CentreLogo
+            );
+
+            return RedirectToAction("Index");
+        }
+
+        private IActionResult EditCentreDetailsPostPreviewSignature(EditCentreDetailsViewModel model)
+        {
+            ClearErrorsForAllFieldsExcept(nameof(EditCentreDetailsViewModel.CentreSignatureFile));
+
+            if (!ModelState.IsValid)
+            {
+                return View(model);
+            }
+
+            if (model.CentreSignatureFile != null)
+            {
+                ModelState.Remove(nameof(EditCentreDetailsViewModel.CentreSignature));
+                model.CentreSignature = imageResizeService.ResizeCentreImage(model.CentreSignatureFile);
+            }
+
+            return View(model);
+        }
+
+        private IActionResult EditCentreDetailsPostRemoveSignature(EditCentreDetailsViewModel model)
+        {
+            ClearErrorsForAllFields();
+
+            ModelState.Remove(nameof(EditCentreDetailsViewModel.CentreSignature));
+            model.CentreSignature = null;
+            return View(model);
+        }
+
+        private IActionResult EditCentreDetailsPostPreviewLogo(EditCentreDetailsViewModel model)
+        {
+            ClearErrorsForAllFieldsExcept(nameof(EditCentreDetailsViewModel.CentreLogoFile));
+
+            if (!ModelState.IsValid)
+            {
+                return View(model);
+            }
+
+            if (model.CentreLogoFile != null)
+            {
+                ModelState.Remove(nameof(EditCentreDetailsViewModel.CentreLogo));
+                model.CentreLogo = imageResizeService.ResizeCentreImage(model.CentreLogoFile);
+            }
+
+            return View(model);
+        }
+
+        private IActionResult EditCentreDetailsPostRemoveLogo(EditCentreDetailsViewModel model)
+        {
+            ClearErrorsForAllFields();
+
+            ModelState.Remove(nameof(EditCentreDetailsViewModel.CentreLogo));
+            model.CentreLogo = null;
+            return View(model);
+        }
+
+        private void ClearErrorsForAllFields()
+        {
+            ClearErrorsForAllFieldsExcept(null);
+        }
+
+        private void ClearErrorsForAllFieldsExcept(string? fieldName)
+        {
+            foreach (var key in ModelState.Keys.Where(k => k != fieldName))
+            {
+                ModelState[key].Errors.Clear();
+                ModelState[key].ValidationState = ModelValidationState.Valid;
+            }
+        }

```

Comment:
I wonder if it's worth writing our own extension methods for this like we do with the User object/claims? Then we can do something like: ModelState.ClearAllErrors() etc